prov/shm/src/smr_ep.c

```diff
@@ -1322,14 +1326,23 @@ static int smr_ep_ctrl(struct fid *fid, int command, void *arg)
 			return ret;
 
 		if (ep->util_ep.caps & FI_HMEM || smr_env.disable_cma) {
-			ep->region->cma_cap_peer = SMR_CMA_CAP_OFF;
-			ep->region->cma_cap_self = SMR_CMA_CAP_OFF;
+			ep->region->cma_cap_peer = SMR_VMA_CAP_OFF;
+			ep->region->cma_cap_self = SMR_VMA_CAP_OFF;
 			if (ep->util_ep.caps & FI_HMEM) {
 				if (ze_hmem_p2p_enabled())
 					smr_init_ipc_socket(ep);
 			}
 		}
 
+#if HAVE_XPMEM
+		if (smr_env.use_xpmem && xpmem)
+			ep->region->xpmem_cap_self = SMR_VMA_CAP_ON;
```

Comment:
Does xpmem have any similar ptrace restrictions we need to check for before enabling?

Reply:
Since xpmem is an out of tree driver, it probably ignores all security concerns.  And may be why it's an out of tree driver...